### PR TITLE
Router updates the url but not the view for launched apps when Go to app event is fired.

### LINKED ISSUE #726: Router updates the url but not the view for launched apps when `Go to app`  event is fired.
### What is the expected behaviour?
In the opened app (viewer mode): when the `Go to app` event is fired we should be updating the view to the targeted app without a full page reload.

### What is the current behaviour?
the router updates the URL but not the view when the `Go to app`  event is fired.

### How to reproduce the issue?

1. add the `Go to app` event to the app, probably to any widget.
2. save and deploy the app
3. click on the launch button to open the app in the viewer in a new browser tab.
4. and trigger the `Go to app` event.

### Screenshots or Screencast

https://user-images.githubusercontent.com/67645175/132842391-55c0cf94-8306-4ed1-b7f6-9793ea6e1c8a.mp4

### PR DESCRIPTION
**Pull request by arpitnath**
Fixes #726 

**App**: https://tooljet-update-pr.herokuapp.com/

**Loom**: https://www.loom.com/share/d127a75817bf449da877d0640ea4510b

### CODE REVIEW

**Review by gondar00 — DISMISSED** [review]

**Comment by arpitnath** [comment]
update: https://www.loom.com/share/e581ad096db74b0c8456c45f9cfc87b4

**Comment by arpitnath** [comment]
### update
app: https://tooljetpr-app.herokuapp.com/apps/5ae21c2a-fd46-4a2b-8b6d-bc805980ae58

File: frontend/src/Editor/Viewer.jsx [hunk 1]
```diff
@@ -33,6 +33,7 @@ class Viewer extends React.Component {
       currentLayout: isMobileDevice ? 'mobile' : 'desktop',
       currentUser: authenticationService.currentUserValue,
       isLoading: true,
+      appLoading: true,
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/Viewer.jsx:36]: what is the difference between `isLoading` and `appLoading`?

File: frontend/src/Editor/Viewer.jsx [hunk 2]
```diff
@@ -150,28 +160,41 @@ class Viewer extends React.Component {
...
+                        <center>
+                          <div className="spinner-border text-azure" role="status"></div>
+                        </center>
+                      </div>
+                    ) : (
+                      <Container
+                        appDefinition={appDefinition}
+                        appDefinitionChanged={() => false} // function not relevant in viewer
+                        snapToGrid={true}
+                        appLoading={isLoading}
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/Viewer.jsx:178]: This line is also confusing, should it be changed to `appLoading={appLoading}`? I think this part of the code requires some refactoring.

File: frontend/src/Editor/Viewer.jsx [hunk 3]
```diff
@@ -150,28 +160,41 @@ class Viewer extends React.Component {
...
-                        onComponentOptionsChanged(this, component, options)
-                      }
-                    />
-                  }
+
+
+                {defaultComponentStateComputed && (
+                  <>
+                    {appLoading ? (
+                      <div className="mx-auto mt-5 w-50 p-5">
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/Viewer.jsx:168]: We have a similar logic at `src/Editor/Container.jsx:315`, why do we need both?

File: frontend/src/Editor/Viewer.jsx (hunk 1 above)
Comment (arpitnath) [reply frontend/src/Editor/Viewer.jsx:36]: `isLoading` if for the currentState. and `appLoading` is for the apps only

**Pull request merged**